server/java/src/main/java/com/godaddy/asherah/grpc/AppEncryptionServer.java

```diff
@@ -0,0 +1,207 @@
+package com.godaddy.asherah.grpc;
+
+import com.godaddy.asherah.appencryption.SessionFactory;
+import com.godaddy.asherah.appencryption.kms.AwsKeyManagementServiceImpl;
+import com.godaddy.asherah.appencryption.kms.KeyManagementService;
+import com.godaddy.asherah.appencryption.kms.StaticKeyManagementServiceImpl;
+import com.godaddy.asherah.appencryption.persistence.DynamoDbMetastoreImpl;
+import com.godaddy.asherah.appencryption.persistence.InMemoryMetastoreImpl;
+import com.godaddy.asherah.appencryption.persistence.JdbcMetastoreImpl;
+import com.godaddy.asherah.appencryption.persistence.Metastore;
+import com.godaddy.asherah.crypto.BasicExpiringCryptoPolicy;
+import com.godaddy.asherah.crypto.CryptoPolicy;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.lang3.SystemUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+import io.grpc.Server;
+import io.grpc.netty.NettyServerBuilder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.kqueue.KQueueServerDomainSocketChannel;
+import io.netty.channel.unix.DomainSocketAddress;
+import picocli.CommandLine;
+
+final class AppEncryptionServer implements Callable<Void> {
+  private static final Logger logger = LoggerFactory.getLogger(AppEncryptionServer.class);
+
+  enum MetastoreType { MEMORY, JDBC, DYNAMODB }
+
+  enum KmsType { STATIC, AWS }
+
+  // Options to configure the metastore
+  @CommandLine.Option(names = "--metastore-type", defaultValue = "MEMORY",
+      description = "Type of metastore to use. Enum values: ${COMPLETION-CANDIDATES}")
+  private static MetastoreType metastoreType;
+  @CommandLine.Option(names = "--jdbc-url",
+      description = "JDBC URL to use for JDBC metastore. Required for JDBC metastore.")
+  private static String jdbcUrl;
+
+  // Options to configure the KMS
+  @CommandLine.Option(names = "--kms-type", defaultValue = "STATIC",
+      description = "Type of key management service to use. Enum values: ${COMPLETION-CANDIDATES}")
+  private static KmsType kmsType;
+  @CommandLine.Option(names = "--preferred-region",
+      description = "Preferred region to use for KMS if using AWS KMS. Required for AWS KMS.")
+  private static String preferredRegion;
+  @CommandLine.Option(names = "--region-arn-tuples", split = ",",
+      description = "Comma separated list of <region>=<kms_arn> tuples. Required for AWS KMS.")
+  private static Map<String, String> regionMap;
+
+  // Options to configure the server
+  @CommandLine.Option(names = "--productId", description = "Specify the product id", required = true)
```

Comment:
command-line options should use consistent casing. rename this to `--product-id`

Reply:
should i call it `product-id` or make it consistent with the `go` parameters which we merged in?

Reply:
`product-id` is fine for now as that would bring it inline with the other flags supported by this implementation. As noted, in our [previous discussion](https://github.com/godaddy/asherah/pull/135#discussion_r405889222) we can worry about cross-implementation consistency later.